hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OmMetadataManagerImpl.java

```diff
@@ -525,22 +525,24 @@ public void start(OzoneConfiguration configuration) throws IOException {
   public static DBStore loadDB(OzoneConfiguration configuration, File metaDir)
       throws IOException {
     return loadDB(configuration, metaDir, OM_DB_NAME, false,
-            java.util.Optional.empty(), true, true);
+            java.util.Optional.empty(), Optional.empty(), true, true);
   }
 
   public static DBStore loadDB(OzoneConfiguration configuration, File metaDir,
                                String dbName, boolean readOnly,
                                java.util.Optional<Boolean>
-                                       disableAutoCompaction)
+                                       disableAutoCompaction,
+                               java.util.Optional<Integer> maxOpenFiles)
```

Comment:
Not sure if `Optional` is preferred here while the parameter could be a primitive `int`?

Also `-1` would be a valid input indicating not limited.

fyi I just realized RocksDB seem to have some usage restrictions when `max_open_files` is not `-1`. Hopefully they don't impact our use case and are unrelated:

1. https://github.com/facebook/rocksdb/blob/cac3240cbfdb0bd4ef1900bc2d664187b3e404bb/db/db_impl/compacted_db_impl.cc#L235-L241
2. https://github.com/facebook/rocksdb/blob/cac3240cbfdb0bd4ef1900bc2d664187b3e404bb/db/version_set.h#L1000-L1002

Reply:
So that validation is happening on the rocksDb side. We shouldn't add another layer of redundant validation. If something is wrong it would just throw exceptions. It is on us to handle it correctly, which I will do as part of HDDS-8202

Reply:
I agree with @smengcl. Someone can set it to zero either by mistake or unintentionally assuming 0 is default.

Reply:
So what should be the expectation here?